gnomad_qc/v3/sample_qc/subpop_analysis.py

```diff
@@ -0,0 +1,217 @@
+
+    # Add sample metadata to the QC MT
+    mt = mt.annotate_cols(**meta_ht[mt.col_key])
+
+    # Filter the  QC MT to the specified pop
+    pop_mt = mt.filter_cols(mt.pop == pop)
+
+    # Apply specified filters
+    if remove_hard_filtered_samples:
+        pop_mt = pop_mt.filter_cols(hl.is_missing(hard_filtered_ht[pop_mt.col_key]))
+    if release:
+        pop_mt = pop_mt.filter_cols(pop_mt.release)
+    if high_quality:
+        pop_mt = pop_mt.filter_cols(pop_mt.high_quality)
+    if outliers is not None:
+        pop_mt = pop_mt.filter_cols(hl.is_missing(outliers[pop_mt.col_key]))
+
+    pop_mt = pop_mt.filter_rows(hl.agg.any(pop_mt.GT.is_non_ref()))
+
+    # Generate a QC MT for the given pop
+    pop_qc_mt = get_qc_mt(
+        pop_mt,
+        min_af=min_af,
+        min_inbreeding_coeff_threshold=min_inbreeding_coeff_threshold,
+        min_hardy_weinberg_threshold=None,
+        ld_r2=None,
+        filter_lcr=False,
+        filter_decoy=False,
+        filter_segdup=False,
+    )
+
+    # Generate PCA data
+    release_string = "_release" if release else ""
+    high_quality_string = "_high_quality" if high_quality else ""
+    ht_path = (
+        get_sample_qc_root()
+        + f"/subpop_analysis/{pop}/{pop}_scores{release_string}{high_quality_string}{outlier_string}.ht"
+    )
+
+    if not (ht_read_if_exists and file_exists(ht_path)):
+        pop_pca_evals, pop_pca_scores, pop_pca_loadings = run_pca_with_relateds(
+            pop_qc_mt, relateds
+        )
+
+        pop_ht = pop_pca_scores.annotate(**meta_ht[pop_pca_scores.key])
+        pop_ht = pop_ht.annotate(
+            project_subpop=pop_ht.project_meta.project_subpop,
+            subpop_description=pop_ht.project_meta.subpop_description,
+            v2_pop=pop_ht.project_meta.v2_pop,
+            v2_subpop=pop_ht.project_meta.v2_subpop,
+            pop=pop_ht.population_inference.pop,
+            project_id=pop_ht.project_meta.project_id,
+            project_pop=pop_ht.project_meta.project_pop,
+        )
+        pop_ht = pop_ht.checkpoint(ht_path, overwrite=True)
+    else:
+        pop_ht = hl.read_table(ht_path)
+
+    return pop_ht
+
+
+def main(args):
+
+    # Read in the raw mt
+    mt = get_gnomad_v3_mt(key_by_locus_and_alleles=True)
+
+    # Filter to test partitions if specified
+    if args.test:
+        logger.info("Filtering to the first two partitions of the MT")
+        mt = mt._filter_partitions(range(2))
+
+    # Write out the densified MT
+    output_path = (
+        get_sample_qc_root()
+        + "/subpop_analysis/"
+        + "gnomad_v3_qc_mt_subpop_analysis.mt"
+    )
+
+    if args.make_subpop_mt:
+        logger.info("Generating densified MT to use for subpop analysis...")
+        mt = compute_subpop_qc_mt(mt, args.min_popmax_af)
+        mt = mt.naive_coalesce(5000)
+        mt = mt.checkpoint(output_path, overwrite=True,)
+
+    logger.info("Generating PCs for subpops...")
+    # TODO: Add code for AMR
```

Comment:
```suggestion
    # TODO: Add code for AMR using currently unused `run_pop_pca` function
```

Reply:
I think it might also be good to add the additional planned parameters with a not implemented error?

Reply:
added planned parameters, what would the not implemented error be?

Reply:
Oh sorry I just meant add a parameter like `--run-subpop-pca` and raise a NotImplementedError for now. It will change, but worth getting the code in for the first step.